### PR TITLE
Fissure Skill

### PR DESCRIPTION
**Pull request by ezrn**
Mostly a copy of fissure from Mineplex with a few major improvement to how it navigates terrain

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested my changes.

### CODE REVIEW

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/mage/axe/Fissure.java [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+    private double fissureExpireDuration;
+    private double damagePerBlock;
+    private int effectDurationIncreasePerLevel;
+    private double damagePerBlockIncreasePerLevel;
+    private int effectDuration;
+    private int slownessLevel;
+    private final Map<Player, ArrayList<Block>> playerFissurePaths = new HashMap<>();
+    private final Map<Player, HashSet<Player>> playerHits = new HashMap<>();
+    private final Map<Player, HashSet<Block>> playerCreatedBlocks = new HashMap<>();
+    private final Map<Player, Integer> fissureExpirationTasks = new HashMap<>();
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/mage/axe/Fissure.java:53]: WeakHashMap

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/mage/axe/Fissure.java [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+            return true;
+        }
+
+        String materialName = material.name().toUpperCase();
+
+        if (materialName.contains("DOOR") || materialName.contains("GATE")) {
+            return true;
+        }
+
+        return false;
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/mage/axe/Fissure.java:348]: block.getState() instanceof Container

Feel like this should be a List loaded from config.

**Review by Mykindos — CHANGES_REQUESTED** [review]
Use WorldBlockHandler to create temporary blocks, they will automatically revert after an amount of time (see Ice Prison for example)

**Comment by ezrn** [comment]
all things done

**Pull request merged**